HealthcareBase/Service/HospitalResourcesService/EquipmentService/EquipmentService.cs

```diff
@@ -69,10 +57,35 @@ private void DeleteFromHospitalizations(EquipmentUnit equipmentUnit)
 
         public void DeleteByType(EquipmentType equipmentType)
         {
-            equipmentType = equipmentTypeRepository.Repository.GetByID(equipmentType.GetKey());
-            foreach (var equipmentUnit in equipmentUnitRepository.Repository.GetAll())
-                if (equipmentUnit.EquipmentType.Equals(equipmentType))
-                    Delete(equipmentUnit);
+           
+        }
+
+        public IEnumerable<EquipmentDto> GetEquipmentByRoomId(int roomId)
+        {
+            return equipmentUnitRepository.Repository.GetColumnsForMatching(
+                condition: equipment => equipment.CurrentLocationId == roomId,
+                selection: equipment => new EquipmentDto()
+                {
+                    Id = equipment.Id,
+                    RoomId = equipment.CurrentLocation.Id,
+                    Name = equipment.EquipmentType.Name
+                }
```

Comment:
Where you define quantity?

Reply:
I have a method in EquipmentService called GetEquipmentWithQuantityByRoomId which calculates the quantity of equipment unit in a particular room.

Reply:
Yes but you are using **uninitialized** quantity from this method and incrementing it.

Reply:
Can i set it to 0 here, and then count it?

Reply:
Yep!